src/main/java/fact/io/hdureader/ZFITSHeapReader.java

```diff
@@ -674,43 +674,52 @@ private Serializable readSingleValueFromBuffer(BinTable.TableColumn c, ByteBuffe
     }
 
     private Serializable readArrayFromBuffer(BinTable.TableColumn c, ByteBuffer buffer) throws IOException {
-
-        if (c.type == BinTable.ColumnType.BOOLEAN) {
-            boolean[] bools = new boolean[c.repeatCount];
-            for (int i = 0; i < c.repeatCount; i++) {
-                bools[i] = buffer.get() > 0;
-            }
-            return bools;
-        }
-
         switch (c.type) {
-            case CHAR:
-                char[] chars = new char[c.repeatCount];
-                buffer.asCharBuffer().get(chars);
-                return chars;
             case BYTE:
                 byte[] b = new byte[c.repeatCount];
                 buffer.get(b);
                 return b;
+            case BOOLEAN:
+                boolean[] bools = new boolean[c.repeatCount];
+                for (int i = 0; i < c.repeatCount; i++) {
+                    bools[i] = buffer.get() == 'T';
+                }
+                return bools;
+            case CHAR:
+                char[] chars = new char[c.repeatCount];
+                for (int i = 0; i < c.repeatCount; i++) {
+                    chars[i] = buffer.getChar();
+                }
+                return chars;
             case SHORT:
                 short[] shorts = new short[c.repeatCount];
-                buffer.asShortBuffer().get(shorts);
+                for (int i = 0; i < c.repeatCount; i++) {
```

Comment:
does this have a performance impact?

Reply:
What's the Performance impact if it was wrong before?

Reply:
About a factor of 6 if I remember correctly. Also to my understanding the `buffer.asShortBuffer().get(shorts);` should be equivalent to the loop. I don't understand where this goes wrong.